tools/importer-rest-api-specs/components/parser/models_test.go

```diff
@@ -1861,3 +1861,20 @@ func TestParseModelWithLocation(t *testing.T) {
 		t.Fatalf("expected example.Fields['Location'] to be a Location but got %q", string(*field.CustomFieldType))
 	}
 }
+
+func TestParseModelBug2675DuplicateModel(t *testing.T) {
+	result, err := ParseSwaggerFileForTesting(t, "models_bug_2675_duplicate_model.json")
+	if err != nil {
+		t.Fatalf("parsing: %+v", err)
+	}
+	if result == nil {
+		t.Fatal("result was nil")
+	}
+
+	example, ok := result.Resources["Example"]
+	if !ok {
+		t.Fatal("the Resource 'Example' was not found")
+	}
+
+	t.Logf("%d", len(example.Operations))
```

Comment:
this commit was intended as a quick POC to prove the problem, would you mind adding a few checks to ensure that we're pulling out the correct number of models/that the models we're expecting actually exist?

Reply:
updated.